authentication.py

```diff
@@ -5,23 +5,49 @@
 
 
 class Authentication:
+	
+	def __init__(self, data_folder: str, password_salt: str) -> None:
+		with open(os.path.join(".", data_folder, "users.json")) as file:
+			self.contents = json.load(file)
+		self.password_salt = password_salt
+		self.data_folder=data_folder
 
-    def __init__(self, data_folder: str, password_salt: str) -> None:
-        with open(os.path.join(".", data_folder, "users.json")) as file:
-            self.contents = json.load(file)
-        self.password_salt = password_salt
+	def is_valid(self, username: str, password: str) -> bool:
+		if username not in self.contents.keys():
+			return False
+		if self._hash_password(password) != self.contents[username]["password"]:
+			return False
+		return True
 
-    def is_valid(self, username: str, password: str) -> bool:
-        if username not in self.contents.keys():
-            return False
-        if self._hash_password(password) != self.contents[username]["password"]:
-            return False
-        return True
+	def user_data(self, username: str) -> Dict:
+		return cast(Dict, self.contents[username])
 
-    def user_data(self, username: str) -> Dict:
-        return cast(Dict, self.contents[username])
+	def _hash_password(self, plaintext_password: str) -> str:
+		hash_algoritm = hashlib.new('sha256')
+		hash_algoritm.update((plaintext_password + self.password_salt).encode("UTF-8"))
+		return hash_algoritm.hexdigest()
 
-    def _hash_password(self, plaintext_password: str) -> str:
-        hash_algoritm = hashlib.new('sha256')
-        hash_algoritm.update((plaintext_password + self.password_salt).encode("UTF-8"))
-        return hash_algoritm.hexdigest()
+	def save(self) -> bool:  # save json to file
+		try:
```

Comment:
Considering the user creation will be triggered from the command line via invoking a file/command, I'd say it is better to let python complain of an I/O error or whatever happens if couldn't save the contents... 
what do you think? (see another comment about prints)